import "proto/base.proto";

package dist_clang.proto;

message Host {
  required string host          = 1;
  optional uint32 port          = 2 [default=6000];
  optional uint32 threads       = 3 [default=2];
  optional bool disabled        = 4 [default=false];
  optional bool only_failed     = 5 [default=false];
}

message Verbosity {
  message Range {
    required uint32 right = 1;
    optional uint32 left  = 2;  // if omitted: [default=right];
  }

  // less'o'equal to |error_mark| levels go to stderr.
  optional uint32 error_mark    = 1 [default=20];

  // all ranges are inclusive: [left, right].
  repeated Range levels         = 2;
}

message Configuration {
  message Cache {
    required string path   = 1;
    optional uint64 size   = 2 [default=0];
    optional bool remote   = 3 [default=true];
    optional bool sync     = 4 [default=false];  // for testing.
    optional bool direct   = 5 [default=false];
    optional bool mtime    = 6 [default=false];  // TODO: not implemented yet.
    optional bool disabled = 7 [default=false];
  }

  optional string socket_path   = 1;
  optional uint32 pool_capacity = 2 [default=16];  // 0 - is unlimited.
  repeated Host remotes         = 4;
  optional Host local           = 5;
  repeated Compiler versions    = 6;
  optional Verbosity verbosity  = 8;
  optional uint32 user_id       = 10;
  optional string config_path   = 13;  // allows reloading of config changes.
  optional Cache cache          = 14;
}
